presentation/src/main/java/com/anytypeio/anytype/presentation/analytics/AnalyticsParamsProvider.kt

```diff
@@ -0,0 +1,35 @@
+package com.anytypeio.anytype.presentation.analytics
+
+import com.anytypeio.anytype.core_models.primitives.SpaceId
+import com.anytypeio.anytype.domain.multiplayer.SpaceViewSubscriptionContainer
+import com.anytypeio.anytype.domain.multiplayer.UserPermissionProvider
+
+interface AnalyticsParamsProvider {
+
+    fun provideParams(spaceId: SpaceId): Params
+
+    data class Params(
+        val permission: String,
+        val spaceType: String
+    )
+}
+
+class DefaultAnalyticsParamsProvider constructor(
+    private val userPermissionProvider: UserPermissionProvider,
+    private val spaceViewContainer: SpaceViewSubscriptionContainer
+) : AnalyticsParamsProvider {
+
+    override fun provideParams(spaceId: SpaceId): AnalyticsParamsProvider.Params {
+        return proceedWithObservingSpaceView(spaceId)
+    }
+
+    private fun proceedWithObservingSpaceView(spaceId: SpaceId): AnalyticsParamsProvider.Params {
```

Comment:
a separate function might be redundant here

Reply:
fixed